validator/src/main/java/com/amazon/aoc/models/kubernetes/KubernetesContextFactory.java

```diff
@@ -0,0 +1,59 @@
+package com.amazon.aoc.models.kubernetes;
+
+import com.amazon.aoc.services.KubernetesService;
+import io.kubernetes.client.openapi.models.V1Pod;
+import java.util.Objects;
+
+public class KubernetesContextFactory {
+  private final String kubeConfigFilePath;
+  private final String deploymentName;
+
+  private final String namespace;
+
+  private final KubernetesService kubernetesService;
+
+  public KubernetesContextFactory(
+      String kubeConfigFilePath, String deploymentName, String namespace) throws Exception {
+    this.kubeConfigFilePath = kubeConfigFilePath;
+    this.deploymentName = deploymentName;
+    this.namespace = namespace;
+    this.kubernetesService = new KubernetesService(this.kubeConfigFilePath);
+  }
+
+  public KubernetesContextFactory(
```

Comment:
nit: since the test class is in the[ same package ](https://github.com/aws-observability/aws-otel-test-framework/pull/1322/files#diff-5f739c96e0ac9d4a16c6a1bc6fceb2b4f2589bbaa730864035f2cc42bec791ebR1)as this one and this constructor is only intended to be used for tests, you can remove the `public` visibility modifier so that this constructor can only be visible in the tests.